### PR TITLE
Incorporated S grade and Additional Courses

### PR DESCRIPTION
**Pull request by dark-trojan**
Both S grade courses and Additional courses are displayed in the course list, and counted towards total credits completed, but aren't considered in the CGPA calculation

### CODE REVIEW

File: src/gpa/activateGPA.js [hunk 1]
```diff
@@ -2,102 +2,95 @@
...
-        if (type_credits_map.has(course.type))
-          type_credits_map.set(course.type, type_credits_map.get(course.type) + course.credits);
-        else
-          type_credits_map.set(course.type, course.credits);
+		if (type_credits_map.has(course.type))
+			type_credits_map.set(course.type, type_credits_map.get(course.type) + course.credits);
+		else type_credits_map.set(course.type, course.credits);
 
+		grade_value = grade_values[course.grade];
+		credits = (course.grade === 'S' || course.type === 'Additional') ? 0 : course.credits;
```
Comment (AdaephonBen) [inline src/gpa/activateGPA.js:75]: @dark-trojan Good work on the PR! Can you please document exactly what's being done here?

File: src/gpa/activateGPA.js [hunk 2]
```diff
@@ -2,102 +2,95 @@
...
+	'A': 10,
+	'A-': 9,
+	'B': 8,
+	'B-': 7,
+	'C': 6,
+	'C-': 5,
+	'D': 4,
+	'FR': 0,
+	'FS': 0,
+	'S': 0
```
Comment (AdaephonBen) [inline src/gpa/activateGPA.js:17]: Also please explain why `'S'` has to be a key in the JSON Object.

**Comment by AdaephonBen** [comment]
Also, more of a general note. The indentation seems to have been affected by whatever formatter you're using, so this PR will have to be put on hold until #18 is solved.

**Comment by npalladium** [comment]
@dark-trojan, your formatter seems to have changed the indentation all over the place. This usually causes problems with `git blame` and causes headaches with merge conflicts when we try to merge other PRs.

File: src/gpa/activateGPA.js (hunk 2 above)
Comment (npalladium) [reply src/gpa/activateGPA.js:17]: I feel that setting S to zero is misleading too. What's your take on this @dark-trojan, @AdaephonBen?

**Comment by npalladium** [comment]
@dark-trojan, we have formatted our code. Could you redo this PR as discussed whenever you are free? Sorry for the inconvenience. 

PS. Take care about the formatting this time.

**Pull request closed without merging**